File: ScanbotSDKexample/Simple UI Components/mrz-scanner/src/main/java/io/scanbot/example/MRZResultActivity.java
package io.scanbot.example;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import io.scanbot.mrzscanner.model.MRZRecognitionResult;

public class MRZResultActivity extends AppCompatActivity {
    public static final String EXTRA_documentCode = "documentCode";
    public static final String EXTRA_firstName = "firstName";
    public static final String EXTRA_lastName = "lastName";
    public static final String EXTRA_issuingStateOrOrganization = "issuingStateOrOrganization";
    public static final String EXTRA_departmentOfIssuance = "departmentOfIssuance";
    public static final String EXTRA_nationality = "nationality";
    public static final String EXTRA_dateOfBirth = "dateOfBirth";
    public static final String EXTRA_gender = "gender";
    public static final String EXTRA_dateOfExpiry = "dateOfExpiry";
    public static final String EXTRA_personalNumber = "personalNumber";
    public static final String EXTRA_optional1 = "optional1";
    public static final String EXTRA_optional2 = "optional2";
    public static final String EXTRA_discreetIssuingStateOrOrganization = "discreetIssuingStateOrOrganization";
    public static final String EXTRA_validCheckDigitsCount = "validCheckDigitsCount";
    public static final String EXTRA_checkDigitsCount = "checkDigitsCount";
    public static final String EXTRA_travelDocTyp = "travelDocType";

    private TextView documentCode;
    private TextView firstName;
    private TextView lastName;
    private TextView issuingStateOrOrganization;
    private TextView departmentOfIssuance;
    private TextView nationality;
    private TextView dateOfBirth;
    private TextView gender;
    private TextView dateOfExpiry;
    private TextView personalNumber;
    private TextView optional1;
    private TextView optional2;
    private TextView discreetIssuingStateOrOrganization;
    private TextView validCheckDigitsCount;
    private TextView checkDigitsCount;
    private TextView travelDocType;

    public static Intent newIntent(Context context, MRZRecognitionResult result) {
        Intent intent = new Intent(context, MRZResultActivity.class);
        intent.putExtra(EXTRA_documentCode, result.documentCode);
        intent.putExtra(EXTRA_firstName, result.firstName);
        intent.putExtra(EXTRA_lastName, result.lastName);
        intent.putExtra(EXTRA_issuingStateOrOrganization, result.issuingStateOrOrganization);
        intent.putExtra(EXTRA_departmentOfIssuance, result.departmentOfIssuance);
        intent.putExtra(EXTRA_nationality, result.nationality);
        intent.putExtra(EXTRA_dateOfBirth, result.dateOfBirth);
        intent.putExtra(EXTRA_gender, result.gender);
        intent.putExtra(EXTRA_dateOfExpiry, result.dateOfExpiry);
        intent.putExtra(EXTRA_personalNumber, result.personalNumber);
        intent.putExtra(EXTRA_optional1, result.optional1);
        intent.putExtra(EXTRA_optional2, result.optional2);
        intent.putExtra(EXTRA_discreetIssuingStateOrOrganization, result.discreetIssuingStateOrOrganization);
        intent.putExtra(EXTRA_validCheckDigitsCount, result.validCheckDigitsCount);
        intent.putExtra(EXTRA_checkDigitsCount, result.checkDigitsCount);
        intent.putExtra(EXTRA_travelDocTyp, result.travelDocType.name());

        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_mrz_result);
        documentCode = (TextView) findViewById(R.id.document_code);
        firstName = (TextView) findViewById(R.id.first_name);
        lastName = (TextView) findViewById(R.id.last_name);
        issuingStateOrOrganization = (TextView) findViewById(R.id.issuingStateOrOrganization);
        departmentOfIssuance = (TextView) findViewById(R.id.departmentOfIssuance);
        nationality = (TextView) findViewById(R.id.nationality);
        dateOfBirth = (TextView) findViewById(R.id.dateOfBirth);
        gender = (TextView) findViewById(R.id.gender);
        dateOfExpiry = (TextView) findViewById(R.id.dateOfExpiry);
        personalNumber = (TextView) findViewById(R.id.personalNumber);
        optional1 = (TextView) findViewById(R.id.optional1);
        optional2 = (TextView) findViewById(R.id.optional2);
        discreetIssuingStateOrOrganization = (TextView) findViewById(R.id.discreetIssuingStateOrOrganization);
        validCheckDigitsCount = (TextView) findViewById(R.id.validCheckDigitsCount);
        checkDigitsCount = (TextView) findViewById(R.id.checkDigitsCount);
        travelDocType = (TextView) findViewById(R.id.travelDocType);

        documentCode.setText(getIntent().getStringExtra(EXTRA_documentCode));
        firstName.setText(getIntent().getStringExtra(EXTRA_firstName));
        lastName.setText(getIntent().getStringExtra(EXTRA_lastName));
        issuingStateOrOrganization.setText(getIntent().getStringExtra(EXTRA_issuingStateOrOrganization));
        departmentOfIssuance.setText(getIntent().getStringExtra(EXTRA_departmentOfIssuance));
        nationality.setText(getIntent().getStringExtra(EXTRA_nationality));
        dateOfBirth.setText(getIntent().getStringExtra(EXTRA_dateOfBirth));
        gender.setText(getIntent().getStringExtra(EXTRA_gender));
        dateOfExpiry.setText(getIntent().getStringExtra(EXTRA_dateOfExpiry));
        personalNumber.setText(getIntent().getStringExtra(EXTRA_personalNumber));
        optional1.setText(getIntent().getStringExtra(EXTRA_optional1));
        optional2.setText(getIntent().getStringExtra(EXTRA_optional2));
        discreetIssuingStateOrOrganization.setText(getIntent().getStringExtra(EXTRA_discreetIssuingStateOrOrganization));
        validCheckDigitsCount.setText("Value: " + getIntent().getIntExtra(EXTRA_validCheckDigitsCount, -1));
        checkDigitsCount.setText("Value: " + getIntent().getIntExtra(EXTRA_checkDigitsCount, -1));
        travelDocType.setText(getIntent().getStringExtra(EXTRA_travelDocTyp));

        findViewById(R.id.retry).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

    }
}

File: classic-components-example/mrz-scanner/src/main/java/io/scanbot/example/MRZResultActivity.kt
package io.scanbot.example

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import io.scanbot.genericdocument.entity.MRZ
import io.scanbot.sdk.mrzscanner.MrzScannerResult

class MRZResultActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_mrz_result)

        val travelDocType = findViewById<TextView>(R.id.travelDocType)
        val documentNumber = findViewById<TextView>(R.id.document_number)
        val firstName = findViewById<TextView>(R.id.first_name)
        val lastName = findViewById<TextView>(R.id.last_name)
        val issuingStateOrOrganization = findViewById<TextView>(R.id.issuingStateOrOrganization)
        val departmentOfIssuance = findViewById<TextView>(R.id.departmentOfIssuance)
        val nationality = findViewById<TextView>(R.id.nationality)
        val dateOfBirth = findViewById<TextView>(R.id.dateOfBirth)
        val gender = findViewById<TextView>(R.id.gender)
        val dateOfExpiry = findViewById<TextView>(R.id.dateOfExpiry)
        val personalNumber = findViewById<TextView>(R.id.personalNumber)
        val optional1 = findViewById<TextView>(R.id.optional1)
        val optional2 = findViewById<TextView>(R.id.optional2)

        val checkDigitBirthDate = findViewById<TextView>(R.id.checkDigitBirthDate)
        val checkDigitDocumentNumber = findViewById<TextView>(R.id.checkDigitDocumentNumber)
        val checkDigitExpiryDate = findViewById<TextView>(R.id.checkDigitExpiryDate)
        val checkDigitGeneral = findViewById<TextView>(R.id.checkDigitGeneral)
        val checkDigitPersonalNumber = findViewById<TextView>(R.id.checkDigitPersonalNumber)

        val dateOfIssuance = findViewById<TextView>(R.id.dateOfIssuance)
        val documentTypeCode = findViewById<TextView>(R.id.documentTypeCode)
        val languageCode = findViewById<TextView>(R.id.languageCode)
        val pinCode = findViewById<TextView>(R.id.pinCode)
        val travelDocTypeVariant = findViewById<TextView>(R.id.travelDocTypeVariant)
        val versionNumber = findViewById<TextView>(R.id.versionNumber)

        val result: MrzScannerResult = intent.getParcelableExtra(EXTRA_MRZ_RESULT)!!

        val mrzResult = MRZ(result.document!!)

        travelDocType.text = mrzResult.travelDocType?.value?.text
        documentNumber.text = mrzResult.documentNumber?.value?.text
        firstName.text = mrzResult.givenNames?.value?.text
        lastName.text = mrzResult.surname?.value?.text
        issuingStateOrOrganization.text = mrzResult.issuingAuthority?.value?.text
        departmentOfIssuance.text = mrzResult.officeOfIssuance?.value?.text
        nationality.text = mrzResult.nationality?.value?.text
        dateOfBirth.text = mrzResult.birthDate?.value?.text
        gender.text = mrzResult.gender?.value?.text
        dateOfExpiry.text = mrzResult.expiryDate?.value?.text
        personalNumber.text = mrzResult.personalNumber?.value?.text
        optional1.text = mrzResult.optional1?.value?.text
        optional2.text = mrzResult.optional2?.value?.text

        checkDigitBirthDate.text = if (mrzResult.checkDigitBirthDate?.isValid == true) "Valid" else "NOT Valid"
        checkDigitDocumentNumber.text = if (mrzResult.checkDigitDocumentNumber?.isValid == true) "Valid" else "NOT Valid"
        checkDigitExpiryDate.text = if (mrzResult.checkDigitExpiryDate?.isValid == true) "Valid" else "NOT Valid"
        checkDigitGeneral.text = if (mrzResult.checkDigitGeneral?.isValid == true) "Valid" else "NOT Valid"
        checkDigitPersonalNumber.text = if (mrzResult.checkDigitPersonalNumber?.isValid == true) "Valid" else "NOT Valid"

        dateOfIssuance.text = mrzResult.dateOfIssuance?.value?.text
        documentTypeCode.text = mrzResult.documentTypeCode?.value?.text
        languageCode.text = mrzResult.languageCode?.value?.text
        pinCode.text = mrzResult.pinCode?.value?.text
        travelDocTypeVariant.text = mrzResult.travelDocTypeVariant?.value?.text
        versionNumber.text = mrzResult.versionNumber?.value?.text

        findViewById<View>(R.id.retry).setOnClickListener { finish() }
    }

    companion object {
        private const val EXTRA_MRZ_RESULT = "MRZ_RESULT"
        @JvmStatic
        fun newIntent(context: Context?, result: MrzScannerResult?): Intent {
            val intent = Intent(context, MRZResultActivity::class.java)
            intent.putExtra(EXTRA_MRZ_RESULT, result)
            return intent
        }
    }
}
